client/data/timeline/hooks.ts

```diff
@@ -0,0 +1,66 @@
+/** @format */
+/**
+ * External dependencies
+ */
+import { useSelect } from '@wordpress/data';
+import { STORE_NAME } from '../constants';
+
+/**
+ * Internal dependencies
+ */
+import { Timeline } from './types';
+import { Charge } from '../charges/types';
+
+export const useTimeline = (
+	chargeId: string
+): {
+	timeline?: Timeline;
+	timelineError?: Error;
+	isLoading: boolean;
+} =>
+	useSelect(
+		( select ) => {
+			const {
+				getTimeline,
+				getTimelineError,
+				isResolving,
+				getCharge,
+				getChargeError,
+			} = select( STORE_NAME );
+
+			// Make sure the charge is loaded first to get the intention ID.
+			const isChargeLoading = isResolving< boolean >( 'getCharge', [
```

Comment:
That's probably a styling preference, but IMO this way it reads a bit better. What do you think?
```suggestion
			const isChargeLoading: boolean = isResolving( 'getCharge', [
```

Reply:
The issue here is more in that the `select()` method returns [generic functions](https://www.typescriptlang.org/docs/handbook/generics.html#hello-world-of-generics) so you must type the function in the angle brackets.

I agree your suggestion reads much better, and your suggestion works for that specific case, but unfortunately doing it that way results in an error on a different line:

<img width="741" alt="image" src="https://user-images.githubusercontent.com/13835680/109343048-b2d35d00-78a7-11eb-8fb9-2c3b56df4103.png">

whereas typing the function in the angle bracket provides the right result:

<img width="956" alt="image" src="https://user-images.githubusercontent.com/13835680/109343166-dac2c080-78a7-11eb-9bb2-ca55b21f6768.png">

I used angle brackets everywhere for the sake of consistency in the code here 🙂

Reply:
Good point, agree!